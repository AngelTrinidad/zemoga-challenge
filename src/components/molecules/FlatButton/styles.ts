import {StyleSheet} from 'react-native';

import {scaleSpace} from '~helpers';

const styles = StyleSheet.create({
  text: {
    marginVertical: scaleSpace(8),
  },
});

export default styles;
