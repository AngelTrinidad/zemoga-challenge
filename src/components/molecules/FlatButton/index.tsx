import React, {FC, ReactNode} from 'react';
import {TextStyle, TouchableOpacity, TouchableOpacityProps} from 'react-native';

import styles from './styles';

import Typography, {TypographyVariant} from '~/components/atoms/Typography';
import {useTheme} from '~contexts';

interface Props extends TouchableOpacityProps {
  children: ReactNode;
  textVariant?: TypographyVariant;
}

const FlatButton: FC<Props> = ({children, textVariant = 'xl', ...props}) => {
  const {colors} = useTheme();

  const textStyle: TextStyle = {
    color: colors.primary,
  };

  return (
    <TouchableOpacity {...props}>
      <Typography variant={textVariant} style={[styles.text, textStyle]}>
        {children}
      </Typography>
    </TouchableOpacity>
  );
};

export default FlatButton;
