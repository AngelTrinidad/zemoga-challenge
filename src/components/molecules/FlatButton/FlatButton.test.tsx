import React from 'react';
import {Text} from 'react-native';

import FlatButton from '.';

import {fireEvent, render} from '~helpers/testing';

describe('<FlatButton />', () => {
  let component: any;
  const onPressMock = jest.fn();

  beforeEach(() => {
    component = render(
      <FlatButton onPress={onPressMock}>
        <Text>Tap it</Text>
      </FlatButton>,
    );
  });

  it('should has same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should call onPress callback on tapped it', () => {
    const button = component.getByText('Tap it');
    fireEvent(button, 'press');
    expect(onPressMock).toBeCalled();
  });
});
