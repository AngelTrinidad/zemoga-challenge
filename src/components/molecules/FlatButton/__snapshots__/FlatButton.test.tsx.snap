// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlatButton /> should has same snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      {
        "color": "#567189",
        "fontFamily": "System",
        "fontSize": 24.57142857142857,
        "marginVertical": 9.142857142857142,
        "textAlign": undefined,
      }
    }
  >
    <Text>
      Tap it
    </Text>
  </Text>
</View>
`;
